src/httpserver.cpp

```diff
@@ -477,6 +475,10 @@ void StopHTTPServer()
         event_base_free(eventBase);
         eventBase = nullptr;
     }
+    if (workQueue) {
+        delete workQueue;
+        workQueue = nullptr;
+    }
```

Comment:
It can't be 😄  this is the goal of this PR - destroy the work queue after event base finish. After `g_thread_http.join()` there can be stuff pending on the event base, and if you delete the `workQueue` before the event base finishes then all pending `workItems` will be destroyed (because they are `unique_ptr`) and then it's only problems from here on.

Reply:
Ah, okay, I didn't understand about the event base; so maybe we could do all the shutdown (deconstruct) of the `workQueue` down here, where you now have the `delete`. Would this work?
<details>
<summary>suggestion</summary>

```
void StopHTTPServer()
{
    LogPrint(BCLog::HTTP, "Stopping HTTP server\n");
    // Unlisten sockets, these are what make the event loop running, which means
    // that after this and all connections are closed the event loop will quit.
    for (evhttp_bound_socket *socket : boundSockets) {
        evhttp_del_accept_socket(eventHTTP, socket);
    }
    boundSockets.clear();
    if (eventBase) {
        LogPrint(BCLog::HTTP, "Waiting for HTTP event thread to exit\n");
        if (g_thread_http.joinable()) g_thread_http.join();
    }
    if (eventHTTP) {
        evhttp_free(eventHTTP);
        eventHTTP = nullptr;
    }
    if (eventBase) {
        event_base_free(eventBase);
        eventBase = nullptr;
    }
    if (workQueue) {
        LogPrint(BCLog::HTTP, "Waiting for HTTP worker threads to exit\n");
        for (auto& thread: g_thread_http_workers) {
            thread.join();
        }
        g_thread_http_workers.clear();
        delete workQueue;
        workQueue = nullptr;
    }
    LogPrint(BCLog::HTTP, "Stopped HTTP server\n");
}
```
</details>

(That seems slightly simpler and clearer to me, but the way you have it now is good.) If you make changes, I don't mind taking another look (and testing) and quickly giving another ack.